enhancements/scheduling/load-aware-vm-rebalancing-through-descheduler.md

```diff
@@ -0,0 +1,472 @@
+
+To increase the leverage against over-utilized nodes, the taint controller can be configured to apply multiple soft taints based on overall resource pressure through `devMultiSoftTainting` option. For example, when a node is classified as over-utilized and its utilization exceeds 50% of the remaining capacity, a second soft taint could be applied. The scaling of taints can follow different models, such as linear or exponential, depending on the desired behavior.
+
+In certain situations, it may be beneficial to evict multiple pods per node within a single descheduling cycle. By exposing `devMultiEvictions` with predefied values of `Simple` (1), `Modest` (2) and `Rapid` (5), one can explore this direction further.
+
+Currently, there is no dependency between the descheduler and the scheduler, so no additional validation of either configuration is required. However, when the scheduler configuration is extended with load-aware elements, the descheduler operator is expected to scale the descheduler down to zero if an invalid combination of configurations is detected.
+
+### Workflow Description
+
+On the descheduler side:
+1. The node classification thresholds gets configured as static (using `thresholds` and `targetThresholds`, as configured in the current `LowNodeUtilization` plugin) or dynamic, based on the state of other nodes in the cluster (using the `useDeviationThresholds` option within the plugin).
+1. The `LowNodeUtilization` plugin classifies nodes into three categories: under-utilized, appropriately-utilized, and over-utilized, as it does today. The metric used for classification will be derived from a PromQL query executed against a Prometheus instance. Prometheus will handle the aggregation of historical data and time series processing. The query is expected to return a scalar [0-1] value for each node.
+1. The `LowNodeUtilization` plugin will keep identifying pods for eviction. A single or multiple pods with the lowest priority are evicted during each descheduling cycle.
+
+On the soft-tainter controller side:
+1. The soft-tainter controller reads the same Prometheus metrics as the descheduler to generate the updated node classification. This ensures that both components are aligned in their understanding of node utilization and can act accordingly based on the latest metrics.
+1. Looping through the node list:
+    1. The soft-tainter controller will apply a soft taint (`effect: PreferNoSchedule`) to the node if it is not already present and the node is classified as overutilized.
+    1. The soft-tainter controller will remove the soft taint from the node if it is present and the node is now classified as either low or appropriately utilized.
+
```

Comment:
should we mention the multi-soft taints logic here?
eg. 2 soft taints for overutilized nodes, 1 for appropriately utilized, 0 for under utilized.
We know that the scheduler is counting the number of soft-taints while weighing the nodes so this will lead under utilized nodes to be preferred over  appropriately utilized that will also be preferred over over utilized ones.

Reply:
Done.